[studio-ui] Dialogs consistency and composition refactors

aachavarria opened the pull request:
- DialogHeader renaming onDismiss to onCloseButtonClick;
- Creating Dialog.tsx component;
- Using Dialog.tsx on CreateFolderDialog.tsx, HistoryDialog.tsx, NewContentDialog.tsx, PublishDialog.tsx;

https://github.com/craftercms/craftercms/issues/4717

aachavarria committed:
Creating Dialog component

aachavarria committed:
DialogHeader renaming onDismiss to onCloseButtonClick;
Creating Dialog.tsx component;
Using Dialog.tsx on CreateFolderDialog.tsx, HistoryDialog.tsx, NewContentDialog.tsx, PublishDialog.tsx;

aachavarria committed:
Moving NewContentDialog to its own folder;
Refactoring NewContentDialog.tsx: styles, translations.ts, utils.ts

rart commented on ui/app/src/components/Dialog/Dialog.tsx:
```diff
@@ -0,0 +1,105 @@
+
+        cancelUnsubscribe = createCustomDocumentEventListener(idCancel, () => {
+          unsubscribe();
+        });
```
```suggestion
        const customEventId = 'dialogDismissConfirm';
        dispatch(
          showConfirmDialog({
            title: formatMessage(translations.pendingChanges),
            onOk: batchActions([dispatchDOMEvent({ id: customEventId, type: 'success' }), closeConfirmDialog()]),
            onCancel: batchActions([dispatchDOMEvent({ id: customEventId, type: 'cancel' }), closeConfirmDialog()])
          })
        );
        unsubscribe = createCustomDocumentEventListener(customEventId, ({ type }) => {
          (type === 'success') && dialogProps.onClose(e, reason);
        });
```

aachavarria committed:
- updating publishg dialog hasPendingChanges when editing submission content
- refactor on showConfirmDialog to use customEventId and send a type

aachavarria committed:
Fixes on CreateFolderDialog.tsx when submitting to set hasPendingChanges and isSubmitting to false;

aachavarria committed:
Refactor on CreateFolderDialog.tsx to its own folder, creating utils.ts, translations.ts;

aachavarria committed:
creating onCloseButtonClick

aachavarria committed:
Moving DependenciesDialog.tsx to its own folder;
Refactoring DependenciesDialog.tsx: translations.ts, utils.ts, interfaces, reducer

aachavarria committed:
adding default exporting to components

aachavarria committed:
Moving HistoryDialog to its own folder;
Refactoring HistoryDialog.tsx: translations.ts, utils.ts, interfaces

aachavarria committed:
Interfaces adjustments to import from DialogProps

aachavarria committed:
Moving ViewVersionDialog.tsx to its own folder;
Refactoring ViewVersionDialog.tsx: translations.ts, utils.ts
Pending to refactor interfaces and add the new Dialog component

aachavarria committed:
Refactoring ViewVersionDialog.tsx.tsx: interfaces

aachavarria committed:
Moving CompareVersionsDialog.tsx to its own folder;
Refactoring CompareVersionsDialog.tsx: translations.ts, utils.ts, interfaces

aachavarria committed:
Creating index.ts on VersionList.tsx

aachavarria committed:
Rollback using new Dialog components on CompareVersionsDialog.tsx, ConfirmDialog.tsx, DependenciesDialog.tsx, NewContentDialog.tsx

rart committed:
Enhanced dialog component development and sample usages, various dialog-related refactors

aachavarria committed:
Updating DeleteDialog.tsx: interfaces, utils, dialog wrapper

aachavarria committed:
Merge branch 'feature/4717' of https://github.com/aachavarria/studio-ui into feature/4717

rart committed:
Remove unused imports in GlobalDialogManager.tsx

aachavarria committed:
renaming Dialog to EnhancedDialog

aachavarria committed:
Merge branch 'feature/4717' of https://github.com/aachavarria/studio-ui into feature/4717

aachavarria committed:
Implementing EnhancedDialog on HistoryDialog.tsx;

aachavarria committed:
Refactoring on HistoryDialog.tsx, CompareVersionsDialog.tsx, ViewVersionDialog.tsx

aachavarria committed:
Refactoring on PublishDialog.tsx

aachavarria committed:
Refactoring NewContentDialog.tsx to use EnhancedDialog

aachavarria committed:
Refactoring CreateFolderDialog.tsx to use EnhancedDialog

aachavarria committed:
Refactoring DeleteDialog.tsx to use EnhancedDialog

aachavarria committed:
Refactoring DependenciesDialog.tsx to use EnhancedDialog

aachavarria committed:
Moving SingleItemSelector.tsx to its own folder;
Moving ContentTypesFilter.tsx to its own folder;
Refactoring changeContentTypeDialog.tsx to use EnhancedDialog

aachavarria committed:
Refactoring WorkflowCancellationDialog.tsx to use EnhancedDialog

aachavarria committed:
Using spread on dialogs state in GlobalDialogManager.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4717

aachavarria committed:
Refactoring RejectDialog.tsx to use EnhancedDialog.tsx
Moving to its own folder;

aachavarria committed:
Refactoring CreateFileDialog.tsx to use EnhancedDialog.tsx
Moving CreateFileDialog to its own folder;

aachavarria committed:
Refactoring UploadDialog.tsx to use useEnhancedDialogState
Moving UploadDialog to its own folder;

aachavarria committed:
updating FormattedMessage id

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4717

aachavarria committed:
refactoring PreviewDialog.tsx to use EnhancedDialog.tsx;
moving PreviewDialog.tsx to its own folder;

aachavarria committed:
Refactoring EditSiteDialog.tsx to use EnhancedDialog.tsx
Moving EditSiteDialog to its own folder;
Cleaning EditSiteDialog.tsx containers code;

aachavarria committed:
Spreading arguments on GlobalDialogManager.tsx PathSelectionDialog component

aachavarria committed:
renaming PublishingStatusDialogBody to PublishingStatusDialogContainer

aachavarria committed:
renaming PublishingStatusDialogBody to PublishingStatusDialogContainer

aachavarria committed:
Improving logic ot ToolsPanelEmbeddedAppViewButton.tsx to support multiple minimize dialogs;
Refactoring WidgetDialog.tsx to use EnhancedDialog;

aachavarria committed:
removing not used id from widget.ts

aachavarria committed:
Implementing EnhancedDialog on CreateUserDialog.tsx

aachavarria committed:
Refactoring CreateUserDialog.tsx to use EnhancedDialog.tsx;
Adding omitHeader to EnhancedDialog.tsx

aachavarria committed:
Refactoring editUserDialog.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Refactoring EditGroupDialog.tsx.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Refactoring CreateTokenDialog.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Refactoring CopyTokenDialog.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Refactoring CodeEditorDialog.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Renaming MinimizedDialog.ts to MinimizedTab.ts
Deleting not used LegacyCodeEditorDialog.tsx.tsx
Renaming push/pop dialog to push/pop tab

aachavarria committed:
Refactoring LegacyFormDialog.tsx to use EnhancedDialog.tsx;

rart commented on ui/app/src/components/CodeEditorDialog/CodeEditorDialogContainer.tsx:
```diff
@@ -157,7 +153,7 @@ export function CodeEditorDialogContainer(props: CodeEditorDialogContainerProps)
     save(() => {
       setContent(editorRef.current.getValue());
-      onMinimized?.();
+      onMinimize();
```
```suggestion
      onMinimize?.();
```

rart commented on ui/app/src/components/CodeEditorDialog/CodeEditorDialogContainer.tsx:
```diff
@@ -190,15 +186,17 @@ export function CodeEditorDialogContainer(props: CodeEditorDialogContainerProps)
-            onClick: onMinimized
+            onClick: onMinimize
           }
         ]}
```
if there's no onMinimize, this whole block should be UNDEFINED

rart reviewed (changes requested):
Partial Review

rart commented on ui/app/src/components/DependenciesDialog/index.ts:
```diff
@@ -0,0 +1,25 @@
+
+export * from './DependenciesDialogUI';
+
+export * from './DependenciesDialogContainer';
```
export utils too. that's where interfaces are.

rart commented on ui/app/src/components/ChangeContentTypeDialog/index.ts:
```diff
@@ -0,0 +1,21 @@
+
+export * from './ChangeContentTypeDialog';
+
+export * from './ChangeContentTypeDialogContainer';
```
export utils too. that's where interfaces are.

rart commented on ui/app/src/components/CodeEditorDialog/index.ts:
```diff
@@ -17,3 +17,5 @@
 
 export * from './CodeEditorDialog';
+
+export * from './CodeEditorDialogContainer';
```
export utils too. that's where interfaces are.

rart commented on ui/app/src/components/CodeEditorDialog/utils.ts:
```diff
@@ -16,6 +16,36 @@
+import { EnhancedDialogState } from '../../utils/hooks/useEnhancedDialogState';
+import React from 'react';
+
+interface CodeEditorDialogBaseProps {
```
export

rart commented on ui/app/src/components/CompareVersionsDialog/index.ts:
```diff
@@ -0,0 +1,21 @@
+
+export * from './CompareVersionsDialog';
+
+export * from './CompareVersionsDialogContainer';
```
export utils

rart commented on ui/app/src/components/ConfirmDialog/index.ts:
```diff
@@ -0,0 +1,21 @@
+
+export * from './ConfirmDialog';
+
+export * from './ConfirmDialogContainer';
```
export utils

rart commented on ui/app/src/components/CopyTokenDialog/index.ts:
```diff
@@ -17,3 +17,5 @@
 
 export * from './CopyTokenDialog';
+
+export * from './CopyTokenContainer';
```
export utils

rart commented on ui/app/src/components/CreateFileDialog/index.ts:
```diff
@@ -0,0 +1,21 @@
+
+export * from './CreateFileDialog';
+
+export * from './CreateFileDialogContainer';
```
export utils

rart commented on ui/app/src/components/CreateFolderDialog/index.ts:
```diff
@@ -0,0 +1,21 @@
+
+export * from './CreateFolderDialog';
+
+export * from './CreateFolderContainer';
```
export utils

rart commented on ui/app/src/components/CreateTokenDialog/index.ts:
```diff
@@ -18,4 +18,4 @@ export { default } from './CreateTokenDialog';
 export * from './CreateTokenDialog';
 
-export * from './CreateTokenDialogUI';
+export * from './CreateTokenDialogContainer';
```
export utils

rart commented on ui/app/src/components/CreateUserDialog/utils.ts:
```diff
@@ -0,0 +1,31 @@
+/*
```
export utils at index.ts of component

rart commented on ui/app/src/components/DeleteDialog/utils.ts:
```diff
@@ -1,17 +1,54 @@
 import StandardAction from '../../models/StandardAction';
```
export utils at index

rart commented on ui/app/src/components/DependenciesDialog/utils.tsx:
```diff
@@ -0,0 +1,94 @@
+/*
```
No jsx here. Rename to `.ts`

rart commented on ui/app/src/components/Dialog/EnhancedDialog.tsx:
```diff
@@ -0,0 +1,99 @@
+/*
```
Directory name should be `EnhancedDialog` (Dialog => EnhancedDialog)

aachavarria committed:
Refactoring NewRemoteRepositoryDialog.tsx to use EnhancedDialog.tsx;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4717

 Conflicts:
	ui/app/src/components/Dialogs/UploadDialog.tsx

rart commented on ui/app/src/state/actions/dialogs.ts:
```diff
@@ -144,10 +144,11 @@ export const workflowCancellationDialogClosed = /*#__PURE__*/ createAction('WORK
 
 // region Legacy Form
-export const showEditDialog = /*#__PURE__*/ createAction<LegacyFormDialogStateProps>('SHOW_EDIT_DIALOG');
+export const showEditDialog = /*#__PURE__*/ createAction<Partial<LegacyFormDialogStateProps>>('SHOW_EDIT_DIALOG');
```
Problem is partial makes all props optional and it isn't try that they all are. We want the typing system to yell if you try to dispatch without one of the required props. Why did you do this? If there's a prop that's optional, it should be declared as such directly on the props interface.

If needed, there can be an update action where everything is optional.

rart commented on ui/app/src/state/actions/misc.ts:
```diff
@@ -22,7 +22,7 @@ export const batchActions = /*#__PURE__*/ createAction<StandardAction[]>('BATCH_
 
 // region dispatch DOM Event
-export const dispatchDOMEvent = /*#__PURE__*/ createAction<{ id: string }>('DISPATCH_DOM_EVENT');
+export const dispatchDOMEvent = /*#__PURE__*/ createAction<{ id: string; type?: string }>('DISPATCH_DOM_EVENT');
```
```suggestion
export const dispatchDOMEvent = /*#__PURE__*/ createAction<{ id: string } & { [prop: string | number]: any }>('DISPATCH_DOM_EVENT');
```

rart commented on ui/app/src/state/reducers/dialogs/delete.ts:
```diff
@@ -47,7 +49,7 @@ export default createReducer<GlobalState['dialogs']['delete']>(initialState, {
     ...payload
   }),
-  [closeDeleteDialog.type]: (state) => ({ ...state, open: false, disableQuickDismiss: false }),
+  [closeDeleteDialog.type]: (state) => ({ ...state, open: false, isSubmitting: false }),
```
Shouldn't we be resetting them all? This goes for the other dialog reducers or containers too.

rart commented on ui/app/src/state/reducers/dialogs/minimizedTabs.ts:
```diff
@@ -0,0 +1,50 @@
+
+export const maximizeTab = /*#__PURE__*/ createAction<Payload>('MAXIMIZE_TAB');
+
+export const updateTab = /*#__PURE__*/ createAction<Partial<MinimizedTab> & Payload>('UPDATE_TAB');
```
Rename as pushMinimizedTab, popMinimizedTab, updateMinimizedTab
Is maximizeTab required? Weren't we going to stick to pop?

rart reviewed (changes requested):
Partial review (109 / 212)

rart commented on ui/app/src/components/WidgetDialog/WidgetDialogContainer.tsx:
```diff
@@ -15,38 +15,19 @@
-        ]}
-      />
-      <section className={classes.toolPanelBody}>
+    <section className={classes.toolPanelBody}>
```
this class name is suspicious 🤨

aachavarria replied:
not sure what it means

rart replied:
The class name doesn't make sense.

rart commented on ui/app/src/components/SystemStatus/GlobalDialogManager.tsx:
```diff
@@ -106,7 +105,11 @@ export const useStyles = makeStyles((theme) =>
+      }
     }
   })
 );
```
I think this is no longer used

rart commented on ui/app/src/components/HistoryDialog/Pagination.tsx:
```diff
@@ -0,0 +1,47 @@
+import { paginationStyles } from './HistoryDialog';
+import { PaginationProps } from './utils';
+
+export function Pagination(props: PaginationProps) {
```
Is this any different from the generic "Pagination" component we have somewhere? Could we sub?

aachavarria replied:
I added a TODO

rart commented on ui/app/src/components/LegacyFormDialog/LegacyFormDialog.tsx:
```diff
@@ -0,0 +1,109 @@
+
+  return (
+    <>
+      <Dialog
```
Shouldn't/couldn't this one be Enhanced too?

aachavarria replied:
This dialog don't need the enhanced dialog because we are calling the formEngine onClose function and it is showing the pendingChanges confirmation message.
This is a complicated dialog and i suggest to keep it using <Dialog at least until we migrate it to not be a legacy wrapper

rart commented on ui/app/src/components/PublishingStatusDialog/PublishingStatusDialog.tsx:
```diff
@@ -17,10 +17,12 @@
 import Dialog from '@material-ui/core/Dialog';
```
No EnhancedDialog?

aachavarria committed:
[4124][next] Bulk upload dialog becomes unresponsive with large volume of files

aachavarria committed:
Creating commonDialogProps util;
Adding export default to dialogs components
Exporting utils from dialog components

aachavarria committed:
removing Partial from action type

aachavarria committed:
exporting utils on deleteDialog
